demos/yielding/README.md

```diff
@@ -0,0 +1,112 @@
+# Yielding
+
+In this demo, we try to show the usage of yielding, which is one characteristic of a multitasking system. 
+
+By calling syscall `yield()`, the thread gives away the utilization of a CPU to another thread, or a specific thread (will be discussed later in another demo).
+
+To achieve thread yielding, we need to introduce the following functions:
+
+* `add_thread_from_worker` (in `src/user/thread.c`)
+* `do_scheduler` (in `src/kernel/thread.c`)
+* `yield` (in `src/user/thread.c`)
```

Comment:
I wonder if these can be links to the files in the repo, but leave that out for now.
(I think they'd need to be markdown links to do that)